src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -835,9 +835,24 @@ trait IterableOps[+A, +CC[X], +C] extends Any {
     *                 corresponding elements of this $coll and `that`. The length
     *                 of the returned collection is the minimum of the lengths of this $coll and `that`.
     */
-  def zip[B](xs: Iterable[B]): CC[(A @uncheckedVariance, B)] = fromIterable(View.Zip(toIterable, xs))
+  def zip[B](xs: Iterable[B]): CC[(A @uncheckedVariance, B)] = zipWith(xs)((_, _))
+
   // sound bcs of VarianceNote
```

Comment:
This comment is really related to the `@uncheckedVariance` above, you should not insert a new line in between them.

Reply:
ooops....will do